### ISSUE TITLE
VM sizes

### ISSUE THREAD

**Issue by polichtm** [MicrosoftLearning/20533-ImplementingMicrosoftAzureInfrastructureSolutions#339]
Pls note that there are changes to the way the VM size selection is handled during execution of the provisioning script. Rather than prompting for the actual size designation (sometimes multiple times during a single deployment), the script prompts once to select the size that has been verified as available by the instructor from a pre-created list and this size is used for the duration of the provisioning script. The same approach applies to all modules that include a scripted VM deployment. The list of VM sizes that the script presents is based on the content of Configfiles\VMSizes.txt - so you can customize it any way you prefer before starting the labs.
Hope this helps - any constructive feedback is always welcome
rgds
Marcin

**Comment by RPCMoritz** [comment]
I'm sitting this course currently, so some feedback for future trainers using this: Windows machines need to have at least 2 G of RAM, and the subscription supports at most 10 cores.
As a general rule, maybe it makes sense to restrict the list of VMSizes to those with 2GB of RAM or more, and 2 cores of vcpu or less. This way, deployment issues are less likely to be experienced, with no real downsides for the course.

**Comment by RPCMoritz** [comment]
True, but for the scope of this course, I think it might be slightly excessive. Maybe it makes sense to add note indicating this to the respective VM-sizes. This reduces the load for trainers, when it comes to providing guidance for suitable VM-sizing.

**Comment by tiagocostapt** [comment]
Yes, Don't take the flexibility of the VM sizes. I have teach this a few days ago with no issues whatever. Regarding the 10 core limit in the Azure Pass, you can open a support ticket and ask an increase into the sub for 20. Even with 10 you can do all the labs without any issues.

**Comment by polichtm** [comment]
Hello Tiago,
appreciate the input
rgds
Marcin
